static/app/components/searchSyntax/parser.tsx

```diff
@@ -1043,11 +1043,16 @@ export const defaultConfig: SearchConfig = {
   },
 };
 
-const options = {
-  TokenConverter,
-  TermOperator,
-  FilterType,
-};
+function tryParseSearch<T extends {config: SearchConfig}>(
+  query: string,
+  config: T
+): ParseResult | null {
+  try {
+    return grammar.parse(query, config);
+  } catch (e) {
+    return null;
```

Comment:
TODO may still be somewhat relevant. Never followed up here

Reply:
Added captureException 🥳